src/tests/Common/XUnitWrapperGenerator/XUnitWrapperGenerator.cs

```diff
@@ -112,40 +131,68 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
                     // and that we shouldn't generate a no-op Main method.
                     return;
                 }
 
-                bool isMergedTestRunnerAssembly = configOptions.GlobalOptions.IsMergedTestRunnerAssembly();
-                configOptions.GlobalOptions.TryGetValue("build_property.TargetOS", out string? targetOS);
-
-                if (isMergedTestRunnerAssembly)
-                {
-                    if (targetOS?.ToLowerInvariant() is "ios" or "iossimulator" or "tvos" or "tvossimulator" or "maccatalyst" or "android" or "browser")
-                    {
-                        context.AddSource("XHarnessRunner.g.cs", GenerateXHarnessTestRunner(methods, aliasMap, assemblyName));
-                    }
-                    else
-                    {
-                        context.AddSource("FullRunner.g.cs", GenerateFullTestRunner(methods, aliasMap, assemblyName));
-                    }
-                }
-                else
-                {
-                    context.AddSource("SimpleRunner.g.cs", GenerateStandaloneSimpleTestRunner(methods, aliasMap));
-                }
+                AddRunnerSource(context, methods, configOptions, aliasMap, compData);
             });
     }
 
+    private static void AddRunnerSource(SourceProductionContext context, ImmutableArray<ITestInfo> methods, AnalyzerConfigOptionsProvider configOptions, ImmutableDictionary<string, string> aliasMap, CompData compData)
+    {
+        bool isMergedTestRunnerAssembly = configOptions.GlobalOptions.IsMergedTestRunnerAssembly();
+        configOptions.GlobalOptions.TryGetValue("build_property.TargetOS", out string? targetOS);
+        string assemblyName = compData.AssemblyName;
+
+        if (isMergedTestRunnerAssembly)
+        {
+            if (targetOS?.ToLowerInvariant() is "ios" or "iossimulator" or "tvos" or "tvossimulator" or "maccatalyst" or "android" or "browser")
+            {
+                context.AddSource("XHarnessRunner.g.cs", GenerateXHarnessTestRunner(methods, aliasMap, assemblyName));
+            }
+            else
+            {
+                context.AddSource("FullRunner.g.cs", GenerateFullTestRunner(methods, aliasMap, assemblyName));
+            }
+        }
+        else
+        {
+            context.AddSource("SimpleRunner.g.cs", GenerateStandaloneSimpleTestRunner(methods, aliasMap));
+        }
+    }
+
+    private static void CheckNoEntryPoint(SourceProductionContext context, CompData compData)
+    {
+        foreach (IMethodSymbol entryPoint in compData.PossibleEntryPoints)
+        {
+            context.ReportDiagnostic(Diagnostic.Create(
+                new DiagnosticDescriptor(
+                    "XUW1001",
+                    "No explicit entry point",
+                    "Projects in merged tests group should not contain entry points",
+                    "XUnitWrapperGenerator",
+                    DiagnosticSeverity.Warning,
+                    isEnabledByDefault: true),
```

Comment:
We should save this in a static readonly field (makes it easier to know which diagnostic ids are in use).

Reply:
Thanks. just fyi - I also noticed that I was generating SimpleRunners for all tests now and restricted the actual wrapper generation to ConsoleApplications (hopefully nothing is depending on libraries having entry points).  I think I will run outerloop+extraplatforms to verify.